api/python/provisioner/commands/check.py

```diff
@@ -609,6 +616,177 @@ def _cluster_status(*, args: str) -> CheckEntry:
+            err_res.set_fail(checked_target=cfg.ALL_MINIONS,
+                             comment="Package cortx-py-utils not installed")
+            res.append(err_res)
+            return res
+
+        check_value : CheckEntry = CheckEntry(cfg.Checks.STORAGE_LUNS.value)
+        try:
+            nodes = Check._get_pillar_data("cluster/node_list")
+            available_luns_checks = (cfg.LUNS_CHECKS)
+
+            for luns_check in available_luns_checks:
+                check_res : CheckEntry = CheckEntry(
+                                         cfg.Checks.STORAGE_LUNS.value)
+                try:
+                    lun_args = [luns_check] + nodes
+                    StorageV().validate('luns', lun_args)
+                    check_res.set_passed(checked_target=cfg.ALL_MINIONS,
+                                         comment="LUNs Accessibility, " 
+                                         "Volume Size and Mapping "
+                                         f"Validated: {lun_args}")
+
+                except Exception as exc:
+                    check_res.set_fail(checked_target=cfg.ALL_MINIONS,
+                                       comment=str(exc))
+                res.append(check_res)
+        except Exception as exc:
+            check_value.set_fail(checked_target=cfg.ALL_MINIONS,
+                                 comment=str(exc))
+            res.append(check_value)
+
+        return res
+
+    @staticmethod
+    def _storage_hba(*, args: str) -> List[CheckEntry]:
+        """Check storage HBA
+        storage_hba <provider_name> <nodes>
+        """
+
+        res: List[CheckEntry] = list()
+        provider = [cfg.HBA_PROVIDER]
+
+        if cortx_py_utils_import_error:
+            err_res : CheckEntry = CheckEntry(cfg.Checks.STORAGE_HBA.value)
+            err_res.set_fail(checked_target=cfg.ALL_MINIONS,
+                             comment="Package cortx-py-utils not installed")
+            res.append(err_res)
+            return res
+
+        check_value : CheckEntry = CheckEntry(cfg.Checks.STORAGE_HBA.value)
+        try:
+            nodes = Check._get_pillar_data("cluster/node_list")
+            check_res : CheckEntry = CheckEntry(cfg.Checks.STORAGE_HBA.value)
+            try:
+                hba_args = provider + nodes
+                StorageV().validate('hba', hba_args)
+                check_res.set_passed(checked_target=cfg.ALL_MINIONS,
+                                     comment="HBA Presence and Ports "
+                                     f"Validated: {hba_args}")
+
+            except Exception as exc:
+                check_res.set_fail(checked_target=cfg.ALL_MINIONS,
+                                   comment=str(exc))
+            res.append(check_res)
+        except Exception as exc:
+            check_value.set_fail(checked_target=cfg.ALL_MINIONS,
+                                 comment=str(exc))
+            res.append(check_value)
+
+        return res
```

Comment:
The same remarks as for `_network_drivers`

Reply:
Comment addressed.

Reply:
The same can be simplified as following:
```python
@staticmethod
def _storage_hba(*, args: str) -> Union[CheckEntry, List[CheckEntry]]:
    """
    Check if Storage HBA is proper
    :param args: storage_hba check specific parameters and arguments
    :return:
    """
    check_value: CheckEntry = CheckEntry(cfg.Checks.STORAGE_HBA.value)
    if cortx_py_utils_import_error:
        check_value.set_fail(checked_target=cfg.ALL_MINIONS,
                         comment="Package cortx-py-utils not installed")
        return check_value

    try:
        nodes = Check._get_pillar_data("cluster/node_list")
    except Exception as exc:
        check_value.set_fail(checked_target=cfg.ALL_MINIONS,
                             comment=str(exc))
        return check_value

    res: List[CheckEntry] = list()
    for provider in cfg.HBA_PROVIDER:
        check_res: CheckEntry = CheckEntry(cfg.Checks.STORAGE_HBA.value)
        try:
            hba_args = [provider] + nodes
            StorageV().validate('hba', hba_args)
        except Exception as exc:
            check_res.set_fail(checked_target=cfg.ALL_MINIONS,
                               comment=str(exc))
        else:
            check_res.set_passed(checked_target=cfg.ALL_MINIONS,
                                 comment="HBA Presence and Ports "
                                         f"Validated: {hba_args}")
        res.append(check_res)

    return res
```

Reply:
Thanks. Comment addressed for methods wherever nested `try-except` was used.